misc/constant_fold.py

```diff
@@ -0,0 +1,19 @@
+# https://github.com/taichi-dev/taichi/pull/839#issuecomment-626217806
+import taichi as ti
+ti.init(print_ir=True)
+#ti.core.toggle_advanced_optimization(False)
```

Comment:
Is this file some kind of demo of turning on/off adavanced_opt? Like the `listgen_demo.py`

Reply:
Yes, it shows off the power of JIT constant folding pass.

Reply:
> Yes, it shows off the power of JIT constant folding pass.

Then probably uses `demo` as the suffix as well? Actually, to keep consistency with the existing `test_*`, `benchmark_*` files, maybe name them as `demo_*` can be more clear.

Reply:
> Then probably uses `demo` as the suffix as well?

Sounds good!